### ISSUE TITLE
Add ability to bookmark the current application state via GC2

### ISSUE THREAD

**Issue by sashuk** [sashuk/vidi#45]

**Comment by sashuk** [comment]
Plan:

1. define interactions with GC2
2. define access mode for module (for authorized and unauthorized users)
3. tracking browsers
4. link sharing and getting rid of URL-passed settings
5. define conflict resolution strategy (when local bookmarks meet the stored bookmarks and vice-versa)
6. develop prototype wireframe

**Comment by sashuk** [comment]
The State snapshot manager prototype:

https://docs.google.com/drawings/d/1yGE_CZo1LHRD3IQ8o3NXWWPT4WkTdPz941z7o5hxxGk/edit?usp=sharing

**Comment by mapcentia** [comment]
Looks good

**Comment by sashuk** [comment]
@mapcentia Is it possible to implement some kind of API described above in order to work with snapshots?

**Comment by mapcentia** [comment]
Yes, I'll implement it in GC2. Could you start by creating an internal API i Node, which stores the snapshot in files? I can when se how it works and create a REST API in GC2.

**Comment by sashuk** [comment]
@mapcentia Sure!

**Comment by sashuk** [comment]
@mapcentia Please check out the https://github.com/sashuk/vidi/blob/ffd84bef0f68476bed9dea8f2e8ba72ad39e77ac/controllers/gc2/stateSnapshots.js
